### PR TITLE
Optimize DDLGuard to remove useless two-level mapping

### LINKED ISSUE #7636: Optimize DDLGuard
## Enhancement
we don't need the two-level mapping `database -> table -> exception_message` for DDLGuard, due to the table(with table_id and keyspace_id) is global unique.

Thus, we should optimize the DDLGuard to simplify the code.

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: close #7636 

Problem Summary:

### What is changed and how it works?
We don't need the two-level mapping database -> table -> exception_message for DDLGuard, because the table(with table_id and keyspace_id) is a global unique.
Thus, we optimize it to simplify the code.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/InterpreterDropQuery.cpp [hunk 1]
```diff
@@ -110,10 +110,8 @@ BlockIO InterpreterDropQuery::execute()
 
         if (table)
             tables_to_drop.emplace_back(table,
-                                        context.getDDLGuard(
-                                            database_name,
-                                            drop.table,
-                                            "Table " + database_name + "." + drop.table + " is dropping or detaching right now"));
+                                        context.getDDLGuard(drop.table,
+                                                            "Table " + database_name + "." + drop.table + " is dropping or detaching right now"));
```
Comment (Lloyd-Pottiger) [inline dbms/src/Interpreters/InterpreterDropQuery.cpp:114]: use` fmt::format` instead

File: dbms/src/Interpreters/InterpreterRenameQuery.cpp [hunk 2]
```diff
@@ -110,16 +110,13 @@ BlockIO InterpreterRenameQuery::execute()
...
                                                      "Table " + from.database_name + "." + from.table_name + " is being renamed right now"));
 
         if (!table_guards.count(to))
             table_guards.emplace(to,
-                                 context.getDDLGuard(
-                                     to.database_name,
-                                     to.table_name,
-                                     "Some table right now is being renamed to " + to.database_name + "." + to.table_name));
+                                 context.getDDLGuard(to.table_name,
+                                                     "Some table right now is being renamed to " + to.database_name + "." + to.table_name));
```
Comment (Lloyd-Pottiger) [inline dbms/src/Interpreters/InterpreterRenameQuery.cpp:119]: ditto

**Review by lidezhu — APPROVED** [review]
LGTM

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**